### PR TITLE
Enumerables to Development branch

### PR DESCRIPTION
**Pull request by glowreeyah**
## In this project  I:

- Implemented the algorithms using Ruby.
- created a number of enumerable methods such as  
my_each
my_each_with_index
my_select
my_all?
my_any?
my_none?
my_count
my_map
my_inject
- tested the methods with the examples I found on the official [Ruby documentation](https://ruby-doc.org/core-2.6.5/Enumerable.html) page.

#### Here is a [demo link](https://repl.it/@glowreeyah/Enumerables) containing the methods I created and their various test cases.

### CODE REVIEW

File: enumerable.rb [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+module Enumerable
+  def my_each
+    return to_enum unless block_given?
+
+    n = 0
+    while n <= size - 1
+      yield(to_a[n])
+      n += 1
+    end
+  end
```
Comment (shshamim63) [inline enumerable.rb]: after calling the given block once for each element in self, returns the array itself. 
So after the line no 10 please add a new line with `self`

File: enumerable.rb [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+
+  def my_each_with_index
+    return to_enum unless block_given?
+
+    n = 0
+    while n <= size - 1
+      yield(to_a[n], n)
+      n += 1
+    end
+  end
```
Comment (shshamim63) [inline enumerable.rb]: after calling the given block once for each element in self, returns the array itself.
So after the line no 20 please add a new line with `self`

**Review by shshamim63 — APPROVED** [review]
## Status :+1: Approved
___
Hi @glowreeyah :wave: ,
Amazing job. You have covered the my_inject(difficult comparing to others) on your first go. Just two simple issues. Check the comments below. As the issues are simple I am approving it in advance. just make the following change and then merge it.
___
#### This :+1: is from me in advance. Don't forget to make the following changes. If you need an explanation don't hesitate to knock me.
___
## :herb: Happy Coding :herb: 
### Shakhawat Hossain
[Linkedin](https://www.linkedin.com/in/shakhawathossainshamim/)
___

File: enumerable.rb (hunk 1 above)
Comment (glowreeyah) [reply enumerable.rb]: Done!

File: enumerable.rb (hunk 2 above)
Comment (glowreeyah) [reply enumerable.rb]: Done!

**Pull request merged**